Plugins/NativeCamera/Source/CameraDevice.h

```diff
@@ -1,12 +1,52 @@
 #pragma once
 
+#include <set>
 #include <bgfx/bgfx.h>
 #include <napi/napi.h>
 #include <arcana/threading/task.h>
+#include <gsl/gsl>
 #include "Capability.h"
 
 namespace Babylon::Plugins
 {
+    enum class RedEyeReduction
+    {
+        Never,
+        Always,
+        Controllable,
+    };
+
+    enum class FillLightMode
+    {
+        Off,
+        Flash,
+        Auto,
+    };
+
+    class PhotoCapabilities final
+    {
+    public:
+        Babylon::Plugins::RedEyeReduction RedEyeReduction{};
+        std::set<Babylon::Plugins::FillLightMode> FillLightModes{};
+
+        int32_t MinWidth{};
+        int32_t MaxWidth{};
+        int32_t StepWidth{};
+
+        int32_t MinHeight{};
+        int32_t MaxHeight{};
+        int32_t StepHeight{};
```

Comment:
I wouldn't expect so, but existing code dealing with width/height was already using `int32_t` (and it is also the numeric type returned from the iOS APIs), so I just matched what was there. Conceptually it could be uint32_t I think, but if we go that route we should change it everywhere and perhaps have code to deal with the possibility that platform APIs we are calling do return a negative value.